packages/ui-app/src/util/isValidBalance.ts

```diff
@@ -0,0 +1,97 @@
+  }
+
+  if (matchEPlus && matchEPlus.length === 1 && !matchEPlusBeforeNumber) {
+    return { isValid: false, errorMessage: 'Exponential \'e+\' notation when used must be followed by a number' };
+  }
+
+  if (matchE && matchE.length === 1 && !matchEBeforeNumberOrPlus) {
+    return { isValid: false, errorMessage: 'Scientific notation \'e\' when used must be followed by a number' };
+  }
+
+  // check the string only contains integers digits or scientific notation
+  if (!reValidInputChars.test(input)) {
+    return { isValid: false, errorMessage: 'Balance to transfer in DOTs must be a number or expressed in scientific notation (i.e. 3.4e38) or exponential with \'e+\'' };
+  }
+
+  // check value is greater than or equal to one, whether it be scientific notation, exponential (which allow decimal)
+  if (isNaN(parseFloat(input)) || Number(parseFloat(input)) < 1) {
+    return { isValid: false, errorMessage: 'Balance to transfer in DOTs must be greater than or equal to one' };
+  }
+
+  // check value is not finite (infinite)
+  if (!isFinite(parseFloat(input))) {
+    return { isValid: false, errorMessage: 'Balance to transfer in DOTs must not be infinite' };
+  }
+
+  // if there is a full stop '.' (only allowed for scientific notation) but they have not yet entered an 'e', then generate error until 'e' provided
+  if (input.indexOf('.') !== -1 && input.indexOf('e') === -1) {
+    return { isValid: false, errorMessage: 'Decimals points are only allowed in scientific notation by using an \'e\' (i.e. 3.4e38) or exponential with \'e+\'' };
+  }
+
+  // chain specification 'latest' is 128 bit and supports below 2^128-1, which is ~3.40×10^38
+  // and may be entered as 3.4e38, 3.4e+38 or 340282366920938463463374607431768211455
+  const supportedBitLength = '128';
+  // show is equivalent to 340282366920938463463374607431768211455 by calling .toString(10)
+  const maxBN128Bit = new BN('2').pow(new BN(supportedBitLength)).sub(new BN('1'));
+  const inputBN = new BN(input);
+
+  if (!inputBN.lt(maxBN128Bit)) {
+    return { isValid: false, errorMessage: 'Balance exceeds maximum for 128 bit' };
+  }
+
+  if (input.indexOf('e') !== -1) {
+    const { inputConvertedFromScientificNotation, errorMessage } = scientificNotationToNumber(input);
+
+    if (!inputConvertedFromScientificNotation) {
+      return { isValid: false, errorMessage: errorMessage };
+    }
+
+    const inputConvertedFromScientificNotationBN = new BN(inputConvertedFromScientificNotation);
+
+    if (matchEPlus && !inputConvertedFromScientificNotationBN.lt(maxBN128Bit)) {
+      return { isValid: false, errorMessage: 'Balance value after converting from exponential notation exceeds maximum for 128 bit' };
+    } else if (!inputConvertedFromScientificNotationBN.lt(maxBN128Bit)) {
+      return { isValid: false, errorMessage: 'Balance value after converting from scientific notation exceeds maximum for 128 bit' };
```

Comment:
Same comments are earlier in the file - between this and all the preceding lines the same issue occurs. (Formatting & non-translatable messages)

Reply:
I've refactored the code so it returns a translated `errorMessage`, but I'm also now returning with it an untranslated `errorMessageUntranslated` so it doesn't interfere with the unit tests as I had difficulty trying to mock the translate function